chain/x/oracle/keeper/result.go

```diff
@@ -50,3 +50,30 @@ func (k Keeper) AddResult(
 	store.Set(types.ResultStoreKey(id), resultHash)
 	return resultHash, nil
 }
+
+// SetResult set result to the store.
+func (k Keeper) SetResult(ctx sdk.Context, regID types.RequestID, result []byte) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.ResultStoreKey(regID), result)
+}
+
+// GetAllResults returns the list of all results in the store. Nil will be added for skipped results.
+func (k Keeper) GetAllResults(ctx sdk.Context) (results [][]byte) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.ResultStoreKeyPrefix)
+	var previousReqID types.RequestID
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		currentReqID := types.RequestID(sdk.BigEndianToUint64(iterator.Key()[1:]))
+		diffReqIDCount := currentReqID - previousReqID
```

Comment:
let's use `:= int(...)` here then we don't need to do casting in the for statement.

Reply:
fixed